aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1550,18 +1552,18 @@ inline void baddbmm_cpu_kernel(const Tensor& result, const Tensor& self, const T
           auto r2 = r1[i];
           auto s2 = s1[i];
           for (const auto j : c10::irange(js)) {
-            scalar_t &r = r2[j];
+            accscalar_t acc_value = 0;
             if (is_bmm) {
-              r = 0;
               for (const auto k : c10::irange(ks)) {
-                r += s2[k] * m1[k][j];
+                acc_value += s2[k] * m1[k][j];
               }
             } else {
-              r *= beta;
+              acc_value *= beta;
```

Comment:
You might have to handle conflict from this PR: https://github.com/pytorch/pytorch/pull/96086. cc @XiaobingSuper

Reply:
Got it, will rebase this PR after https://github.com/pytorch/pytorch/pull/96086 is merged.